### PR TITLE
New CI image

### PR DESCRIPTION
**Pull request by TriplEight**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ default:
     name: parity-build
 
 .docker-env:                       &docker-env
-  image:                           parity/rust-builder:latest
+  image:                           paritytech/substrate-ci-linux:production
```
Comment (kaichaosun) [inline .gitlab-ci.yml:61]: Devhub's [node-template](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/docker-compose.yml#L6) now use `parity/rust-builder:latest` as a base image when run in docker. Should I change it to the new image?

Comment (TriplEight) [reply .gitlab-ci.yml:61]: I will make a PR there, thanks for the heads up. I know some other projects are using this image and will make sure all of them will use a new one.

**Comment by gavofyork** [comment]
@TriplEight can this really be merged?

**Comment by TriplEight** [comment]
@gavofyork yes, right after https://gitlab.parity.io/parity/infrastructure/scripts/-/jobs/545541 is successful. 

The new `paritytech/ci-linux:staging` image is being tested with `test-linux-stable` and `check-web-wasm` tests before being published as `paritytech/ci-linux:staging` which will be used in Substrate's CI.

**Comment by bkchr** [comment]
bot merge

**Pull request merged**